# Stefan Teleman <stefan.teleman@oracle.com>
# C++ wants const-correctness. NULL isn't const-correct.
--- gcc-5.3.0/libgcc/gthr-posix.h	2015-01-05 04:33:28.000000000 -0800
+++ gcc-5.3.0/libgcc/gthr-posix.h	2015-08-04 09:00:39.627285519 -0700
@@ -659,7 +659,8 @@
 __gthread_create (__gthread_t *__threadid, void *(*__func) (void*),
 		  void *__args)
 {
-  return __gthrw_(pthread_create) (__threadid, NULL, __func, __args);
+  const pthread_attr_t *__pta = 0;
+  return __gthrw_(pthread_create) (__threadid, __pta, __func, __args);
 }
 
 static inline int
@@ -729,7 +730,10 @@
 __gthread_mutex_init_function (__gthread_mutex_t *__mutex)
 {
   if (__gthread_active_p ())
-    __gthrw_(pthread_mutex_init) (__mutex, NULL);
+  {
+    const pthread_mutexattr_t *__pma = 0;
+    __gthrw_(pthread_mutex_init) (__mutex, __pma);
+  }
 }
 
 static inline int
@@ -842,7 +846,10 @@
 __gthread_cond_init_function (__gthread_cond_t *__cond)
 {
   if (__gthread_active_p ())
-    __gthrw_(pthread_cond_init) (__cond, NULL);
+  {
+    const pthread_condattr_t *__pca = 0;
+    __gthrw_(pthread_cond_init) (__cond, __pca);
+  }
 }
 #endif
 
